src/Uno.UI.Runtime.Skia.Wpf/Extensions/Storage/Pickers/FileOpenPickerExtension.cs

```diff
@@ -45,17 +45,31 @@ private async Task<StorageFile[]> OpenPickerAsync(bool multiple)
 			};
 
 			var filterBuilder = new StringBuilder();
-			filterBuilder.Append(string.Join("|", _picker.FileTypeFilter.Select(fileType => $"{fileType}|*{fileType}")));
+			var filterIndex = -1;
+			var fileTypeFilterCount = _picker.FileTypeFilter.Count;
+			for (var i = 0; i < fileTypeFilterCount; i++)
+			{
+				var fileType = _picker.FileTypeFilter[i];
+				if (fileType == "*")
+				{
+					filterBuilder.Append("|All|*");
+					filterIndex = i + 1;
```

Comment:
The `+ 1` here is confusing me. Don't we want the same index as the `*` ?

Reply:
Microsoft.Win32.OpenFileDialog.FilterIndex  is 1-based see [this](https://docs.microsoft.com/it-it/dotnet/api/microsoft.win32.filedialog.filterindex?view=windowsdesktop-6.0#remarks)

Reply:
That would be a great in-code comment.